admin/services/access-arrangements.service.js

```diff
@@ -30,7 +32,16 @@ accessArrangementsService.submit = async (submittedData, dfeNumber, userId) => {
   }
   const pupil = await pupilDataService.sqlFindOneBySlugAndSchool(urlSlug, dfeNumber)
   const processedData = await accessArrangementsService.process(submittedData, pupil, dfeNumber, userId)
-  return accessArrangementsService.save(processedData, pupil)
+  const displayData = await accessArrangementsService.save(processedData, pupil)
+  const results = await pinGenerationDataService.sqlFindActivePinRecordsByUrlSlug(urlSlug)
+  // Sync existing preparedCheck(s) when 1 or more active pins exist
+  if (results.length > 0) {
+    const checkCodes = results.map(r => r.checkCode)
+    checkCodes.forEach(checkCode => {
+      azureQueueService.addMessage('prepared-check-sync', { version: 1, checkCode: checkCode })
```

Comment:
@jon-shipley While adding `await` is a good point and I seem to agree with you, I have to ask.
How come other message submissions(look at pupil-status or prepare-check) don't have that wait as well?
I thought the admin app shouldn't await for the submission to the queue to complete.
If we choose to add the await then catch shouldn't be an issue since the whole implementation is wrapped within a try/catch on controller level.

Reply:
if you don't wait, how do you know the message was received successfully?